lib/rubocop/cop/rspec/variable_name.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # This cop makes sure that all variables use the configured style.
+      #
+      # @example EnforcedStyle: snake_case (default)
+      #   # bad
+      #   let(:userName) { 'Adam' }
+      #   subject(:userName) { 'Adam' }
+      #
+      #   # good
+      #   let(:user_name) { 'Adam' }
+      #   subject(:user_name) { 'Adam' }
+      #
+      # @example EnforcedStyle: camelCase
+      #   # bad
+      #   let(:user_name) { 'Adam' }
+      #   subject(:user_name) { 'Adam' }
+      #
+      #   # good
+      #   let(:userName) { 'Adam' }
+      #   subject(:userName) { 'Adam' }
+      class VariableName < Cop
+        include RuboCop::RSpec::Language::NodePattern
+        include ConfigurableNaming
+
+        MSG = 'Use %<style>s for variable names.'
+
+        def_node_matcher :variable_definition?, <<~PATTERN
+          (send #{RSPEC} #{Variables::ALL.node_pattern_union}
+            ({sym str} $_) ...)
+        PATTERN
+
+        def on_send(node)
+          variable_definition?(node) do |variable_name|
+            check_name(node, variable_name, node.loc.expression)
```

Comment:
perhaps we can highlight only the actual variable name? For this you will have to capture the actual sym/str node, and not only the value (but then it have a `.value` method that you can use), otherwise with the current code you need `node.first_argument.loc.expression`